db-client-java/src/main/java/com/eventstore/dbclient/ExpectedRevision.java

```diff
@@ -5,17 +5,51 @@
 
 import java.util.Objects;
 
+/**
+ * Constants used for expected revision control.
+ * <p>
+ * The use of expected revision can be a bit tricky especially when discussing guaranties given by the
+ * EventStoreDB server.
+ *
+ * The EventStoreDB server will assure idempotency for all requests using any value in <i>ExpectedRevision</i> except
```

Comment:
**SUGGESTION:** I think it may be worth adding a sentence around optimistic concurrency, as idempotency is less important here.

Reply:
Done